rfcs/044-patron-deletions/README.md

```diff
@@ -0,0 +1,25 @@
+# RFC 044: Tracking Patron Deletions
+
+**Status:** Draft
+
+**Last updated:** 3/2/22
+
+## Context
+
+We use Sierra as a source of truth for patron accounts: all account information is held in Sierra as a sole source of truth, and Sierra is used for all credential validation. For us, Auth0 effectively acts as the implementation for all of the OAuth2 roles, on top of Sierra.
+
+While Sierra is the source of truth, Auth0 does store a minimal amount of patron data (email, barcode, patron number). We need to be able to get rid of the records in Auth0 when a record is removed from Sierra, for GDPR reasons as much as anything else.
+
+Sierra does not provide a change feed or any other kind of notification for events, so we need a polling solution. We already do this for items and bibs [in the Sierra adapter](https://github.com/wellcomecollection/catalogue-pipeline/tree/main/sierra_adapter#readme), by polling for changes in a given time window. We do not need the complex architecture of the Sierra adapter for tracking patron deletions: we don't need any information from within the records, and we don't need to merge anything. 
+
+## Proposal
+
+1. A **CloudWatch event** is run at some interval *T* to trigger a Lambda
+2. The **Lambda** queries the Sierra API for patron records<sup>†</sup> that were deleted in the previous time window *W > T*. That is to say, time windows overlap to avoid records getting lost at the boundary.
```

Comment:
Given that the relevant query only has date granularity, I'm not sure this is the right way to think of the Window/Interval relationship.

As long as we regularly run it against completed days, there is no boundary in which things may get lost.

Because the granularity of the data we are looking at is per-day, then the logical frequency would also be daily, e.g. run it in nightly in the wee hours to bring yesterday up to date.

If we do run it against today, e.g. because we want to be more responsive to changes than daily, then we will also need to run the whole day again once it has completed, in order to be sure that we catch everything.  Running against today is always going to be a window anchored on midnight.

Reply:
Ah, good point - hadn't been thinking about it in terms of acting on the previous day, and wanted the overlapping windows (as in the Sierra adapter) because I'd imagined a case where you'd miss deletions that happened on a given day after the service had run. A non-issue if just looking at the previous day - will update.